tools/migration/cmd/go-filecoin-migrate.go

```diff
@@ -0,0 +1,105 @@
+package cmd
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/filecoin-project/go-filecoin/tools/migration/internal"
+	"github.com/filecoin-project/go-filecoin/tools/migration/migrate_1-to-2"
+)
+
+type Migrator interface {
+	Migrate(oldRepo, newRepo *os.File) error
+	Describe()
+	Validate(oldRepo, newRepo *os.File) error
+}
```

Comment:
Blocking: document this interface extensively, including the semantics of the old and new repos. I am still confused about why two repos are provided to `Migrate`: if the new repo is non-empty, then it should be migrated in place and have no opportunity to mutate the original. Otherwise, is the migration responsible for doing the copy?

The name `Migrator` is a bit misleading, not clear whether it's the orchestrator or the migration logic. I suggest `Migration` instead.

I expect `Describe()` to return a string. If not, provide an output writer explicitly rather than assuming anything.

Reply:
I'm confused between your objection to Migrator here vs your request for it below. I'd like to take the below suggestion to move it to migration/migrator.go, however that is where main.go is and go doesn't allow >1 package name at the same level.  So if you prefer moving it, I suggest moving it to `migration/internal/migrator.go` .   The pattern I was attempting to follow, somewhat, is this one: [golang-standards/project-layout](https://github.com/golang-standards/project-layout/tree/master/cmd):

> `/cmd`
> Main applications for this project.
> The directory name for each application should match the name of the executable you want to have (e.g., /cmd/myapp).

Reply:
@anorth I think this has been mostly addressed in #2614 with intention of adding more in a follow-up.